db/crud.go

```diff
@@ -1165,6 +1165,93 @@ func (db *DatabaseCollectionWithUser) PutExistingRevWithBody(ctx context.Context
 
 }
 
+func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Body, docID string) (*uint32, base.Set, channels.AccessMap, channels.AccessMap, error) {
+	doc := &Document{
+		ID:    docID,
+		_body: body,
+	}
+	oldDoc := doc
+	if docID != "" {
+		if docInBucket, err := db.GetDocument(ctx, docID, DocUnmarshalAll); err == nil {
+			oldDoc = docInBucket
+			if doc._body == nil {
+				body = oldDoc._body
+				doc._body = body
+			}
+			doc._body[BodyRev] = oldDoc.SyncData.CurrentRev
+		} else {
+			return nil, nil, nil, nil, err
+		}
+	}
+
+	delete(body, BodyId)
+
+	// Get the revision ID to match, and the new generation number:
+	matchRev, _ := body[BodyRev].(string)
+	generation, _ := ParseRevID(ctx, matchRev)
+	if generation < 0 {
+		return nil, nil, nil, nil, base.HTTPErrorf(http.StatusBadRequest, "Invalid revision ID")
+	}
+	generation++
+	oldDoc.RevID = body.ExtractRev()
```

Comment:
What body is this extracting the rev from?  It looks like it's the body that was passed in - I expected it to be docInBucket.SyncData.CurrentRev.  Do we have unit tests that validate that rev is being properly populated into doc and oldDoc when doing a dry run of the sync function?